src/main/java/org/candlepin/resource/util/ResourceDateParser.java

```diff
@@ -68,7 +68,7 @@ public static Date parseDateString(String activeOn) {
             d = DatatypeConverter.parseDateTime(activeOn).getTime();
         }
         catch (IllegalArgumentException e) {
-            throw new BadRequestException("Invalid date, must use ISO 8601 format", e);
+            throw new BadRequestException("Invalid date, \"" + activeOn + "\" must use ISO 8601 format", e);
```

Comment:
:+1:

Side note, though: if this string is being returned to the user, we should probably be translating it.

Reply:
Done.